### PR TITLE
Add a script to build the testsuite, virtualenv and all

### PR DESCRIPTION
**Pull request by gsnedders**
cc/ @jgraham @plinss

---
This change is [Reviewable](https://reviewable.io/reviews/w3c/csswg-test/1214)

### CODE REVIEW

File: build-css-testsuites.sh [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    if [ -z $PYTHON ]; then
+        echo "Please ensure Python 2 is installed"
+        exit 1
+    fi
+
+    $PYTHON -m virtualenv $VENV || { echo "Please ensure virtualenv is installed"; exit 2; }
+fi
+
+# Install dependencies
+$VENV/bin/pip install six==1.10.0 Template-Python==0.1.post1 html5lib==0.9999999 lxml==3.7.3 mercurial==4.1
```
Comment (jgraham) [inline build-css-testsuites.sh]: Add a requirements.txt file.

File: build-css-testsuites.sh [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+if [ -z $VENV ]; then
+    VENV=tools/_virtualenv
+fi
+
+# Create the virtualenv
+if [ ! -d $VENV ]; then
+    if [ -z $PYTHON ]; then
+        command -v python
+        if [ $? -eq 0 ]; then
+            if [ `python -c 'import sys; print("%i" % (sys.hexversion<0x03000000))'` -eq 1 ]; then
```
Comment (jgraham) [inline build-css-testsuites.sh]: `python -c 'import sys; print(int(sys.version_info[0] < 3))'` seems easier to understand

File: build-css-testsuites.sh [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+    if [ -z $PYTHON ]; then
+        command -v python2
+        if [ $? -eq 0 ]; then
+            PYTHON=python2
+        fi
+    fi
+
+    if [ -z $PYTHON ]; then
+        echo "Please ensure Python 2 is installed"
```
Comment (jgraham) [inline build-css-testsuites.sh]: `2.7`

File: build-css-testsuites.sh (hunk 2 above)
Comment (jgraham) [inline build-css-testsuites.sh]: (and makes it easier to check that it's 2.7)

Comment (gsnedders) [inline build-css-testsuites.sh]: I've just gone with `sys.version[0:3] == "2.7"`

**Comment by jgraham** [comment]
Reviewed 1 of 1 files at r4.
Review status: 1 of 2 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/w3c/csswg-test/1214)*

**Comment by jgraham** [comment]
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/w3c/csswg-test/1214)*

**Pull request merged**